tests/test_api_callable.py

```diff
@@ -141,6 +141,27 @@ def test_call_kwargs(self):
         self.assertEqual(my_callable(None, CallOptions(key='updated')),
                          'updated')
 
+    def test_call_merge_options_metadata(self):
+        settings_kwargs = {
+            'key': 'value',
+            'metadata': [('key1', 'val1'), ('key2', 'val2')]
+        }
+
+        settings = _CallSettings(kwargs=settings_kwargs)
+        my_callable = api_callable.create_api_call(
+            lambda _req, _timeout, **kwargs: kwargs, settings)
+
+        expected_kwargs = settings_kwargs
+        self.assertEqual(my_callable(None), expected_kwargs)
+
+        expected_kwargs = {
+            'key': 'value',
+            'metadata': [('key1', 'updated_val1'), ('key2', 'val2')]
+        }
+        self.assertEqual(
+            my_callable(None, CallOptions(metadata=[('key1', 'updated_val1')])),
```

Comment:
Please not, in real world scenario the the merging logic does not change the ordering for x-goog-client-api (it remains the last header). So, ordering should not impact correctness (because according to the http specification order of headers should not matter), and in real world scenarios ordering remains the same, so it should not impact performance as well.